### PR TITLE
Balanced optimization strategy

### PR DESCRIPTION
**Pull request by matteo-ronchetti**
Adds balanced optimization strategy as described in #38

### CODE REVIEW

File: lambda/finalizer.js [hunk 1]
```diff
@@ -27,11 +29,19 @@ const getStrategy = (event) => {
     return event[0].strategy || defaultStrategy;
 };
 
+const getBalancedWeight = (event) => {
+    // extract weight used by balanced strategy or fallback to default (0.5)
+    const w = ("balancedWeight" in event[0]) ?  event[0].balancedWeight : defaultBalancedWeight;
```
Comment (alexcasalboni) [inline lambda/finalizer.js]: Preferred syntax:

```javascript
const weight = event[0].balancedWeight || defaultBalancedWeight;
```

File: lambda/finalizer.js [hunk 2]
```diff
@@ -88,3 +98,25 @@ const findFastest = (stats) => {
     // just return the first one
     return stats[0];
 };
+
+const findBalanced = (stats, w) => {
```
Comment (alexcasalboni) [inline lambda/finalizer.js]: Please use `weight` instead of `w`. It's just more readabale :)

File: lambda/finalizer.js (hunk 1 above)
Comment (matteo-ronchetti) [reply lambda/finalizer.js]: That was the first thing I tried but later realized that it doesn't work when balancedWeight = 0

File: lambda/finalizer.js (hunk 2 above)
Comment (matteo-ronchetti) [reply lambda/finalizer.js]: Fixed

File: lambda/finalizer.js (hunk 1 above)
Comment (matteo-ronchetti) [reply lambda/finalizer.js]: Fixed

**Comment by alexcasalboni** [comment]
Merged!

I'll publish on AWS SAR as `v2.1.2` asap (likely tomorrow or Monday).

Thanks @matteo-ronchetti 🙏

**Pull request merged**